controller/pkg/extensions/konnect/controlplane_test.go

```diff
@@ -0,0 +1,729 @@
+			},
+			secretExists:   true,
+			expectedCert:   "test-certificate",
+			expectedKey:    "test-private-key",
+			expectedErrMsg: "",
+		},
+		{
+			name:           "error - secret not found",
+			secretExists:   false,
+			expectedCert:   "",
+			expectedKey:    "",
+			expectedErrMsg: "failed to get TLS client secret test-namespace/test-secret",
+		},
+		{
+			name: "error - missing tls.crt",
+			secretData: map[string][]byte{
+				"tls.key": []byte("test-private-key"),
+			},
+			secretExists:   true,
+			expectedCert:   "",
+			expectedKey:    "",
+			expectedErrMsg: "TLS certificate not found in secret test-namespace/test-secret",
+		},
+		{
+			name: "error - missing tls.key",
+			secretData: map[string][]byte{
+				"tls.crt": []byte("test-certificate"),
+			},
+			secretExists:   true,
+			expectedCert:   "",
+			expectedKey:    "",
+			expectedErrMsg: "TLS key not found in secret test-namespace/test-secret",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			builder := fake.NewClientBuilder()
+
+			if tt.secretExists {
+				secret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-secret",
+						Namespace: "test-namespace",
+					},
+					Data: tt.secretData,
+				}
+				builder = builder.WithObjects(secret)
+			}
+
+			cl := builder.Build()
+
+			cert, key, err := getTLSClientCertAndKey(context.Background(), cl, "test-secret", "test-namespace")
+
+			if tt.expectedErrMsg != "" {
+				assert.Error(t, err)
+				assert.Contains(t, err.Error(), tt.expectedErrMsg)
+				assert.Empty(t, cert)
+				assert.Empty(t, key)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expectedCert, cert)
+				assert.Equal(t, tt.expectedKey, key)
+			}
+		})
+	}
+}
+
+func TestGetKonnectConfig(t *testing.T) {
+	tests := []struct {
+		name                   string
+		konnectExtensionConfig *KonnectExtensionConfig
+		expectedConfig         *managercfg.KonnectConfig
+	}{
+		{
+			name:                   "nil KonnectExtensionConfig",
+			konnectExtensionConfig: nil,
+			expectedConfig:         nil,
+		},
+		{
+			name: "valid KonnectExtensionConfig",
+			konnectExtensionConfig: &KonnectExtensionConfig{
+				KonnectConfig: &managercfg.KonnectConfig{
+					Address: "https://example.com",
+				},
+			},
+			expectedConfig: &managercfg.KonnectConfig{
+				Address: "https://example.com",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			processor := &ControlPlaneKonnectExtensionProcessor{
+				KonnectExtensionConfig: tt.konnectExtensionConfig,
+			}
+			result := processor.GetKonnectConfig()
+			assert.Equal(t, tt.expectedConfig, result)
+		})
+	}
+}
+
+func TestControlPlaneKonnectExtensionProcessor_Process(t *testing.T) {
+	// Create a scheme for the fake client.
+	s := scheme.Scheme
+	require.NoError(t, konnectv1alpha1.AddToScheme(s))
+	require.NoError(t, konnectv1alpha2.AddToScheme(s))
+	require.NoError(t, corev1.AddToScheme(s))
```

Comment:
For this you can just use `scheme.Get()` from https://github.com/Kong/kong-operator/blob/0356fc46538454b8d856662de5f979d1b92d6abd/modules/manager/scheme/scheme.go#L21-L43 which will always contain all the necessary schemas (as otherwise all the other code will fail).

This will also prevent the need for updating code here in case the required API versions change.